soyuz_app/views/user.py

```diff
@@ -70,81 +70,86 @@ def signup(request, batch_id, email):
     # get number of students in batch
     num_students_in_batch = batch.users.count()
 
-    if request.method == 'GET':
-        first_name = request.GET.get("first_name", "")
-        last_name = request.GET.get("last_name", "")
-
-        # if difference is more than days env var, registration is not allowed
-        if difference.days < int(days_to_expiration):
-            return render(request, "users/registration-expired.html")
-
-        else:
-            form = SignUpForm(
-                initial={
+    # check to see if user already exists in database
+    try:
+        user = get_user_model().objects.get(email=email)
+    # if user does not exist, render signup form
+    except get_user_model().DoesNotExist:
+        if request.method == 'GET':
+            first_name = request.GET.get("first_name", "")
+            last_name = request.GET.get("last_name", "")
+
+            # if difference is more than days env var, registration is not allowed
+            if difference.days < int(days_to_expiration):
```

Comment:
is the comment saying otherwise? or am i reading something wrongly? code looks right though.
anw i realise this was old code and you didn't actually touch this this round.